Assets/com.gdcbd.bossbattle/Scripts/Player/PlayerController.cs

```diff
@@ -0,0 +1,298 @@
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace com.gdcbd.bossbattle.player
+{
+    [RequireComponent(typeof(Rigidbody2D), typeof(Collider2D))]
+    public class PlayerController : MonoBehaviour
+    {
+        [SerializeField] private PlayerHealth Health;
+
+        [SerializeField]  private PlayerGunManager playerGunManager;
+        [SerializeField] private PlayerVisualController _playerVisualController;
+        [SerializeField] private Transform _gunTransform;
+
+        private Rigidbody2D _rigidbody;
+        private CapsuleCollider2D _colider;
+        private InputManager _input;
+
+        private Vector2 _velocity;
```

Comment:
Maybe we should write a seperate data class to store all these bools? It is not readable

Reply:
Cannot solve this 
those are internal variable to process data
I only can simplify this by Divided Player Controller into different part basis of the task
like HandleGravity, GroundDetector, JumpHandler etc.